nmigen/vendor/intel.py

```diff
@@ -146,6 +146,29 @@ def add_clock_constraint(self, clock, frequency):
         super().add_clock_constraint(clock, frequency)
         clock.attrs["keep"] = "true"
 
+    @property
+    def default_clk_constraint(self):
+        # Internal high-speed oscillator on Cyclone V devices.
+        # It is specified to not be faster than 100MHz, but the actual
+        # frequency seems to vary a lot between devices. Measurements
+        # of 78 to 84 MHz have been observed.
+        if self.default_clk == "cyclonev_oscillator" and self.device.startswith("5C"):
```

Comment:
The check for `self.device.startswith("5C")` should be an assertion, since otherwise the error message you get if you misconfigure the oscillator is less clear.